purchase_order_block/views/purchase_order_view.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 Eficent Business and IT Consulting Services S.L.
+     Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
+
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form.block.reason</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <field name="date_order" position="after">
+                <field name="po_block_id" attrs="{'readonly': [('id','!=', False)], 'invisible':['|', ('released','=', True), ('empty_check','=', True)]}"/>
```

Comment:
I would always show the po_block_id so you can see whether it is set or not

Reply:
We only need:
` <field name="po_block_id" attrs="{'readonly': ['|', ('id','!=', False), ('state', '!=', 'draft')]"/>`

why make it invisible?